### ISSUE TITLE
Not an issue but a question

### ISSUE THREAD

**Issue by Saif-Fazlani** [AlphaWallet/alpha-wallet-ios#4359]
Hi, I'm working on a fork of AlphaWallet for the past four months. Recently, I merged the latest code of AlphaWallet into my fork and unfortunately, it broke some of my functionality.
Previously, as soon a specific token, named FRZN, was sent to my wallet, the notifySubscribersOfUpdatedTokens() in TokenCollection was called, where I extracted that token's balance by filtering on: 
privateTokenObjects, which was set as:
privateTokenObjects = tokenDataStores.compactMap { $0.enabledObject }.flatMap { $0 }, 
and saved it to UserDefaults to use later on in the application.
But now, that function is replaced by a function named fetch(), which contains a single line of code i.e. refereshSubject.send(()) as its definition. I'm unable to figure out where the definition of this function lies and hence unable to extract that token's (FRZN) balance.
Not an issue but a question
Any help would be appreciated, thanks!

**Comment by oa-s** [comment]
i n case i understood your question right, you need to get all tokens of all enabled chains. to filter their balances for later storing:
You can use separate subscription on`tokensDataStore.enabledTokenObjectsChangesetPublisher(forServers: enabledServers)` it returns updates for all tokens. firstly it returns a changeset (enum with cases `initial`, `updates`, `error`), later you can map its values to get an array of tokens.

**Comment by Saif-Fazlani** [comment]
Thanks for your reply.
Let me rephrase my question for a better understanding.

Previously, while an app instance was running, and I sent a new token to my wallet address, the notifySubscribersOfUpdatedTokens() in TokenCollection was called and updated the privateTokenObjects variable (i.e. Added the newly added token into the list) and hence I could easily extract the required information for that token, without relaunching the app.

Now, while following the same scenario, fetch() is called since it has replaced notifySubscribersOfUpdatedTokens() and after it's execution (i.e. refereshSubject.send(())), if I call tokensDataStore.enabledTokenObjectsChangesetPublisher(forServers: enabledServers) OR simply tokensDataStore.enabledTokenObjects(forServers: enabledServers), the newly added token isn't in the list and doesn't get added until I relaunch the app.

**Comment by oa-s** [comment]
to make the same behaviour as was bevore updates you can the next steps: 
- add a new token to TokensDataStore;
- call method `enabledTokenObjects(forServers: enabledServers)`.

**Comment by oa-s** [comment]
can you please share peace of code you are using to add a new token?
actually when add a new token, `ChangesetPublisher` get called automatically, and you should receive a list of tokens with newwelly added token.

**Comment by oa-s** [comment]
@Saif-Fazlani reopen or recreate when have a new questions
